pkg/migrations/sql/preinstall/008-tracing.sql

```diff
@@ -0,0 +1,253 @@
+GRANT SELECT ON TABLE SCHEMA_TRACING.tag_key TO prom_reader;
+GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE SCHEMA_TRACING.tag_key TO prom_writer;
+GRANT USAGE ON SEQUENCE SCHEMA_TRACING.tag_key_id_seq TO prom_writer;
+
+CREATE TABLE SCHEMA_TRACING.tag
+(
+    id BIGINT NOT NULL GENERATED ALWAYS AS IDENTITY,
+    tag_type SCHEMA_TRACING_PUBLIC.tag_type NOT NULL,
+    key_id bigint NOT NULL,
+    key SCHEMA_TRACING_PUBLIC.tag_k NOT NULL REFERENCES SCHEMA_TRACING.tag_key (key) ON DELETE CASCADE,
+    value SCHEMA_TRACING_PUBLIC.tag_v NOT NULL,
+    UNIQUE (key, value) INCLUDE (id, key_id)
+)
+PARTITION BY HASH (key);
+GRANT SELECT ON TABLE SCHEMA_TRACING.tag TO prom_reader;
+GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE SCHEMA_TRACING.tag TO prom_writer;
+GRANT USAGE ON SEQUENCE SCHEMA_TRACING.tag_id_seq TO prom_writer;
+
+-- create the partitions of the tag table
+DO $block$
+DECLARE
+    _i bigint;
+    _max bigint = 64;
+BEGIN
+    FOR _i IN 1.._max
+    LOOP
+        EXECUTE format($sql$
+            CREATE TABLE SCHEMA_TRACING.tag_%s PARTITION OF SCHEMA_TRACING.tag FOR VALUES WITH (MODULUS %s, REMAINDER %s)
+            $sql$, _i, _max, _i - 1);
+        EXECUTE format($sql$
+            ALTER TABLE SCHEMA_TRACING.tag_%s ADD PRIMARY KEY (id)
+            $sql$, _i);
+        EXECUTE format($sql$
+            GRANT SELECT ON TABLE SCHEMA_TRACING.tag_%s TO prom_reader
+            $sql$, _i);
+        EXECUTE format($sql$
+            GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE SCHEMA_TRACING.tag_%s TO prom_writer
+            $sql$, _i);
+    END LOOP;
+END
+$block$
+;
+
+CREATE TYPE SCHEMA_TRACING_PUBLIC.span_kind AS ENUM
+(
+    'SPAN_KIND_UNSPECIFIED',
+    'SPAN_KIND_INTERNAL',
+    'SPAN_KIND_SERVER',
+    'SPAN_KIND_CLIENT',
+    'SPAN_KIND_PRODUCER',
+    'SPAN_KIND_CONSUMER'
+);
+GRANT USAGE ON TYPE SCHEMA_TRACING_PUBLIC.span_kind TO prom_reader;
+
+CREATE TYPE SCHEMA_TRACING_PUBLIC.status_code AS ENUM
+(
+    'STATUS_CODE_UNSET',
+    'STATUS_CODE_OK',
+    'STATUS_CODE_ERROR'
+);
+GRANT USAGE ON TYPE SCHEMA_TRACING_PUBLIC.status_code TO prom_reader;
+
+CREATE TABLE IF NOT EXISTS SCHEMA_TRACING.span_name
+(
+    id bigint NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    name text NOT NULL CHECK (name != '') UNIQUE
+);
+GRANT SELECT ON TABLE SCHEMA_TRACING.span_name TO prom_reader;
+GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE SCHEMA_TRACING.span_name TO prom_writer;
+GRANT USAGE ON SEQUENCE SCHEMA_TRACING.span_name_id_seq TO prom_writer;
+
+CREATE TABLE IF NOT EXISTS SCHEMA_TRACING.schema_url
+(
+    id bigint NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    url text NOT NULL CHECK (url != '') UNIQUE
+);
+GRANT SELECT ON TABLE SCHEMA_TRACING.schema_url TO prom_reader;
+GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE SCHEMA_TRACING.schema_url TO prom_writer;
+GRANT USAGE ON SEQUENCE SCHEMA_TRACING.schema_url_id_seq TO prom_writer;
+
+CREATE TABLE IF NOT EXISTS SCHEMA_TRACING.inst_lib
```

Comment:
i'm gonna be annoying: are we sure we want to shorten instrumentation? I thing the name should be `instrumentation_lib`

Reply:
I'm fine with that. It was originally `instrumentation_library` but that got super annoying to type. `instrumentation_lib` is a good middle ground